pkg/k8s/client.go

```diff
@@ -564,3 +567,42 @@ func (c *Client) GetUpstreamTrafficSetting(namespace *types.NamespacedName) *pol
 	}
 	return nil
 }
+
+// GetMeshRootCertificate returns a MeshRootCertificate resource with namespaced name
+func (c *Client) GetMeshRootCertificate(mrcName string) *configv1alpha2.MeshRootCertificate {
+	key := types.NamespacedName{Namespace: c.osmNamespace, Name: mrcName}.String()
+	resource, exists, err := c.informers.GetByKey(osminformers.InformerKeyMeshRootCertificate, key)
+	if exists && err == nil {
+		return resource.(*configv1alpha2.MeshRootCertificate)
+	}
+	return nil
+}
+
+// UpdateMeshRootCertificate updates a MeshRootCertificate.
+func (c *Client) UpdateMeshRootCertificate(obj *configv1alpha2.MeshRootCertificate) (*configv1alpha2.MeshRootCertificate, error) {
+	return c.configClient.ConfigV1alpha2().MeshRootCertificates(c.osmNamespace).Update(context.Background(), obj, metav1.UpdateOptions{})
+}
+
+// UpdateMeshRootCertificateStatus updates the status of a MeshRootCertificate.
+func (c *Client) UpdateMeshRootCertificateStatus(obj *configv1alpha2.MeshRootCertificate) (*configv1alpha2.MeshRootCertificate, error) {
+	return c.configClient.ConfigV1alpha2().MeshRootCertificates(c.osmNamespace).UpdateStatus(context.Background(), obj, metav1.UpdateOptions{})
+}
+
+// ListMeshRootCertificates returns the MRCs stored in the informerCollection's store
+func (c *Client) ListMeshRootCertificates() ([]*configv1alpha2.MeshRootCertificate, error) {
```

Comment:
suggestion: Let's add another method: `AddMRCEventsHandler`, and remove the informersCollection from the MRCComposer

Reply:
Added!